src/Microsoft.AspNetCore.Blazor.Browser.JS/src/GlobalExports.ts

```diff
@@ -1,16 +1,20 @@
-﻿import { platform } from './Environment'
-import { registerFunction } from './Interop/RegisteredFunction';
-import { navigateTo } from './Services/UriHelper';
-import { invokeDotNetMethod, invokeDotNetMethodAsync } from './Interop/InvokeDotNetMethodWithJsonMarshalling';
+import { platform } from './Environment';
+import { navigateTo, internalFunctions as uriHelperInternalFunctions } from './Services/UriHelper';
+import { internalFunctions as httpInternalFunctions } from './Services/Http';
+import { attachRootComponentToElement, renderBatch } from './Rendering/Renderer';
 
 if (typeof window !== 'undefined') {
   // When the library is loaded in a browser via a <script> element, make the
   // following APIs available in global scope for invocation from JS
   window['Blazor'] = {
     platform,
-    registerFunction,
     navigateTo,
-    invokeDotNetMethod,
-    invokeDotNetMethodAsync
+
+    _internal: {
+      attachRootComponentToElement,
+      renderBatch,
+      http: httpInternalFunctions,
+      uriHelper: uriHelperInternalFunctions
+    }
```

Comment:
I can see that you're doing a relayer on some of the internals of `blazor.js` - I'm mostly ignoring these since I'm not sure I would have any meaningful feedback.

Reply:
So as I understand it - you're making everything was a registered function into a function that's reachable on some path from the root (`window`) since that's the new mechanism we're using to bind functions.